lambda_code/transfer/Server/index.py

```diff
@@ -0,0 +1,88 @@
+    DISABLE_PHYSICAL_RESOURCE_ID_GENERATION = True  # Use Server Id instead
+
+    def validate(self):
+        self.endpoint_type = self.resource_properties.get('EndpointType', PUBLIC_ENDPOINT_TYPE)
+        self.endpoint_details = self.resource_properties.get('EndpointDetails')
+        self.identity_provider_type = self.resource_properties.get('IdentityProviderType', SERVICE_IDENTITY_PROVIDER_TYPE)
+        self.identity_provider_details = self.resource_properties.get('IdentityProviderDetails')
+        self.hostkey = self.resource_properties.get('HostKey')
+        self.logging_role = self.resource_properties.get('LoggingRole')
+
+        if VPC_ENDPOINT_TYPE == self.endpoint_type and self.endpoint_details is None:
+            raise ValueError('{} type specified without endpoint details'.format(VPC_ENDPOINT_TYPE))
+
+        if PUBLIC_ENDPOINT_TYPE == self.endpoint_type and self.endpoint_details is not None:
+            raise ValueError('{} type specified *with* endpoint details'.format(PUBLIC_ENDPOINT_TYPE))
+
+        if API_GATEWAY_IDENTITY_PROVIDER == self.identity_provider_type and self.identity_provider_details is None:
+            raise ValueError('{} type specified without identity provider details'.format(API_GATEWAY_IDENTITY_PROVIDER))
+
+        if SERVICE_IDENTITY_PROVIDER_TYPE == self.identity_provider_type and self.identity_provider_details is not None:
+            raise ValueError('{} type specified *with* identity provider details'.format(SERVICE_IDENTITY_PROVIDER_TYPE))
+
+    @staticmethod
+    def add_if_not_none(params, key, value):
+        if value is not None:
+            params[key] = value
+
+    def build_params(self):
+        required = {
+            'EndpointType': self.endpoint_type,
+            'IdentityProviderType': self.identity_provider_type
+        }
+        self.add_if_not_none(required, 'EndpointDetails', self.endpoint_details)
+        self.add_if_not_none(required, 'IdentityProviderDetails', self.identity_provider_details)
+        self.add_if_not_none(required, 'HostKey', self.hostkey)
+        self.add_if_not_none(required, 'LoggingRole', self.logging_role)
+
+        return required
+
+    def create(self):
+        transfer_client = self.get_boto3_client('transfer')
+
+        params = self.build_params()
+        response = transfer_client.create_server(**params)
+        self.physical_resource_id = response['ServerId']
+
+        return {'ServerId': self.physical_resource_id}
+
+    def update(self):
+        transfer_client = self.get_boto3_client('transfer')
+
+        params = self.build_params()
+        params['ServerId'] = self.physical_resource_id
+        transfer_client.update_server(**params)
+
+        return {'ServerId': self.physical_resource_id}
+
+    def delete(self):
+        transfer_client = self.get_boto3_client('transfer')
+
+        transfer_client.delete_server(ServerId=self.physical_resource_id)
+
+        return {'ServerId': self.physical_resource_id}
```

Comment:
Not wrong, but useless on delete

Reply:
So just return nothing?

Reply:
Yes. Just omit the `return` statement. Or `return None` if you really want to be explicit